templates/common/baremetal/files/NetworkManager-clean-initrd-state.yaml

```diff
@@ -0,0 +1,79 @@
+    # This activation generates NM state which makes that profile to be
+    # re-activated by the NM service. And this has the effect that other
+    # profiles deployed by the user for the same set of devices are ignored.
+
+    # The intention here is to remove the state that was generated with the
+    # activation of those profiles during dracut execution. Then when NM service
+    # runs, the profiles generated by dracut, along with other profiles
+    # deployed, are evaluated towards finding the most appropriate profile to
+    # connect a device with.
+    #
+    # There is an RFE filed against NM that once implemented would make this
+    # script unnecessary:
+    # https://bugzilla.redhat.com/show_bug.cgi?id=2089707
+
+    # check that we are doing this for the expected kernel arguments
+    CMD_LINE_REGEX="(^|\s+)ip=(dhcp|dhcp6|dhcp,dhcp6|dhcp6,dhcp)(\s+|$)"
+    grep -Eqs "$CMD_LINE_REGEX" /proc/cmdline || exit
+
+    for device in "${NM_DEVICES_DIR}"/*; do
+      [ -e "$device" ] || continue
+
+      # get the uuid of the profile the device has been activated with
+      device_profile_uuid=$(sed -nr '/^\[device\]/,/^\[/{/^connection-uuid\s*=/{s/[^=]+\s*=\s*//;P}}' "$device")
+
+      # the device was not activated with any profile, ignore
+      [ -n "$device_profile_uuid" ] || continue
+
+      # find the profile by uuid
+      for profile in "${NM_CONN_DIR}"/*; do
+        profile_uuid=$(sed -nr '/^\[connection\]/,/^\[/{/^uuid\s*=/{s/[^=]+\s*=\s*//;P}}' "$profile")
+        [ "$profile_uuid" = "$device_profile_uuid" ] && break
+      done
+
+      # profile not found, ignore
+      [ "$profile_uuid" = "$device_profile_uuid" ] || continue
+
+      # profile not generated by nm-initrd-generator, ignore
+      # only check if the key is set (from NM 1.32.4)
+      origin=$(sed -nr '/^\[user\]/,/^\[/{/^org.freedesktop.NetworkManager.origin\s*=/{s/[^=]+\s*=\s*//;P}}' "$profile")
+      [ -z "$origin" ] || [ "$origin" = "nm-initrd-generator" ] || continue
+
+      # the device file name is the ifindex
+      ifindex=$(basename "$device")
+      
+      # get the interface name by ifindex
+      ifname=$(ip -j link show | jq -r ".[] | select(.ifindex == ${ifindex}) | .ifname // empty")
+
+      # no interface name found, ignore
+      [ -n "$ifname" ] || continue
+
+      logger -t nm-clean-initrd-state "$ifname activated with nm-init-rd-generator profile, removing state..."
+      
+      # NM can still generate internal profiles from the IP address
+      # configuration of devices, so flush addresses
+      logger -t nm-clean-initrd-state "Flushing IP addresses from $ifname"
+      ip addr flush "$ifname"
+      ip -6 addr flush "$ifname"
+
+      # remove device state file to prevent NM to uniletarelly connect with the
+      # latest activated profile without evaluating other profiles 
+      logger -t nm-clean-initrd-state "Removing $device"
+      rm -f "$device"
```

Comment:
maybe I'm paranoid but I like a `--` with `rm -f`

`rm -f -- "$device"`

but if someone puts a `-foo` in `/run/NetworkManager/devices` then other weird things happen?